halo/app/ready.go

```diff
@@ -0,0 +1,96 @@
+package app
+
+import (
+	"encoding/json"
+	"io"
+	"sync"
+
+	"github.com/omni-network/omni/lib/errors"
+)
+
+// Consist of all flags describing the current readiness status of a halo node.
+type readinessStatus struct {
+	// Indicates if the current halo instance uses an EVM RPC endpoint instead of running an execution client.
+	peersExposed bool
+	mu           sync.RWMutex
+	status       flags
+}
+
+type flags struct {
+	ConsensusSynced    bool   `json:"consensus_synced"`      // Unhealthy if false
+	ConsensusP2PPeers  int    `json:"consensus_p_2_p_peers"` // Unhealthy if 0
+	ExecutionConnected bool   `json:"execution_connected"`   // Unhealthy if false
```

Comment:
we should probably update `ExecutionConnected` if we loose connection?

Reply:
It looks like the only way how the disconnect manifests itself in the app is when we cannot read peer count or the block number. So I can set the disconnection status upon getting any of these two errors. Do you see a better way doing this?

Reply:
yeah, if either of those error, set connected=false